ui/apps/dashboard/src/app/(dashboard)/env/[environmentSlug]/apps/[externalID]/syncs/Sync/FunctionList.tsx

```diff
@@ -31,57 +49,111 @@ export function FunctionList({ className, removedFunctions, syncedFunctions }: P
   }, [syncedFunctions]);
 
   return (
-    <div className={classNames('grid grid-cols-2 border border-slate-300 bg-white', className)}>
-      <div className="flex border-b border-r border-slate-300 p-2">
-        <CheckCircleIcon className="mr-1 text-green-600" height={20} />
-        <h2>Synced Functions ({syncedFunctions.length})</h2>
-      </div>
-      <div className="flex border-b border-slate-300 p-2">
-        <MinusCircleIcon className="mr-1 text-red-600" height={20} />
-        <h2>Removed Functions ({removedFunctions.length})</h2>
-      </div>
+    <>
+      <CollapsibleCardRoot
+        type="single"
+        defaultValue={openSyncedFunctions ? 'syncedFunctions' : undefined}
+        collapsible
+      >
+        <CollapsibleCardItem value="syncedFunctions">
+          <CollapsibleCardHeader className="flex items-center justify-between border-slate-300 px-6 py-3 text-sm font-medium text-slate-600 data-[state=open]:border-b">
+            <h2>Synced Functions ({syncedFunctions.length})</h2>
+            <CollapsibleCardTrigger
+              asChild
+              onClick={() => setSyncedFunctions(!openSyncedFunctions)}
```

Comment:
```suggestion
              onClick={() => setSyncedFunctions((prev) => !prev)}
```

When changing state based on the current state, using a reducer can prevent some race condition bugs